lib/deviantart_gallery_downloader.rb

```diff
@@ -27,27 +28,35 @@ def fetch
 
     folders.each do |folder|
       image_page_links = get_image_page_links(folder[:link])
-      image_page_links.each_with_index do |page_link, index|
-        retry_count = 0
-        begin
-          @agent.get(page_link)
-          download_button_link = @agent.page.parser.css(".dev-page-button.dev-page-button-with-text.dev-page-download").map{|a| a["href"]}[0]
-          image_link = @agent.page.parser.css(".dev-content-full").map{|img| img["src"]}[0]
-          if ARGV[0].include?("-s")
-            download_link = image_link
-          else
-            download_link = download_button_link || image_link
-          end
-          file_path = get_file_path(index, image_page_links, download_link, folder[:name])
-          @agent.get(download_link).save(file_path) unless File.exist?(file_path)
-        rescue => ex
-          puts ex.message
-          if retry_count < 3
-            retry_count += 1
-            puts "retrying..."
-            retry
-          else
-            next "failed after 3 retries, next"
+      if not image_page_links.nil?
```

Comment:
This and the other `if` block can be replaced by `return if`, better for reading, I think
```ruby
return if image_page_links.nil?

# carry on image fetching
```

Reply:
Resolved in d8550a6